packages/strapi-tiptap-editor/admin/src/utils/generateLabel.ts

```diff
@@ -0,0 +1,40 @@
+import truncate from 'lodash.truncate';
+import striptags from 'striptags';
+
+interface GenerateLabelData {
+  name: string;
+  content: string;
+}
+
+export const generateLabel = ({ name, content }: GenerateLabelData): { label: string; name: string } | null => {
+  if (!content?.trim()) return null;
+
+  // Preserve new lines by replacing block elements with \n before stripping tags
+  const formattedContent = content.replace(/<\/(p|div|h[1-6]|li|blockquote|tr|td)>/gi, '\n');
+
+  // Strip HTML tags but preserve new lines
+  const plainText = striptags(formattedContent)
+    .replace(/(\s|&nbsp;)+/g, ' ') // Replace any whitespace or non-breaking space with a single space
+    .trim(); // Trim the result to remove leading/trailing spaces
+
+  // Truncate the text safely without cutting words
+  const label = truncate(plainText, {
+    length: 70, // Max length of 50 characters
+    separator: /,? +/, // Cut at spaces or commas
+  });
+
+  // Generate a unique key using the field name
+  const labelKey = `richtextContent_${name}`;
+  try {
+    // Dispatch a custom event to notify about the label update
+    window.dispatchEvent(
+      new CustomEvent('labelUpdated', {
+        detail: { key: labelKey, label, name, value: content },
+      }),
+    );
+
+    return { label, name };
+  } catch (error) {
+    return null;
+  }
+};
```

Comment:
Wat denk je ervan om `generateLabel` en `dispatchLabel` als losse functies te hebben? De ene is generic, de andere is specifiek voor Strapi.